pkg/dashboard/core/common.go

```diff
@@ -82,19 +84,19 @@ func (f *Filter) ConstructQueryArgs() (string, []interface{}) {
 	}
 
 	startEnd := ""
-	if f.Start != "" && f.End != "" {
+	if f.Start != zeroTime && f.End != zeroTime {
 		startEnd = "created_at BETWEEN ? AND ?"
 		args = append(args, f.Start, f.End)
-	} else if f.Start != "" && f.End == "" {
+	} else if f.Start != zeroTime && f.End == zeroTime {
 		startEnd = "created_at >= ?"
 		args = append(args, f.Start)
-	} else if f.Start == "" && f.End != "" {
+	} else if f.Start == zeroTime && f.End != zeroTime {
 		startEnd = "created_at <= ?"
 		args = append(args, f.End)
 	}
 
 	if startEnd != "" {
-		if len(args) > 0 {
+		if len(query) > 0 {
```

Comment:
I think there is no need to modify it here? That is to say the same as line 76.

Reply:
This is because if we only pass start or end, the `args` with bigger than zero. So the query result will be like `AND created_at >= xxx`, this will generate an incorrect statement that will cause the SQL query to fail.

Reply:
> I think there is no need to modify it here? That is to say the same as line 76.

The previous is fine because we append `query` and `args` at the same time but at the `startEnd` we have pre-append `args`. This causes us to need to determine whether the `query` is empty or not.

Reply:
okay